core/src/main/java/com/sequenceiq/cloudbreak/service/user/RemoteUserDetailsService.java

```diff
@@ -84,39 +84,27 @@
     @Override
     @Cacheable(value = "userCache", key = "#filterValue")
     public CbUser getDetails(String filterValue, UserFilterField filterField) {
-
-        HttpHeaders tokenRequestHeaders = new HttpHeaders();
-        tokenRequestHeaders.set("Authorization", getAuthorizationHeader(clientId, clientSecret));
-
-        Map<String, String> tokenResponse = restTemplate.exchange(
-                identityServerUrl + "/oauth/token?grant_type=client_credentials",
-                HttpMethod.POST,
-                new HttpEntity<>(tokenRequestHeaders),
-                Map.class).getBody();
-
-        HttpHeaders scimRequestHeaders = new HttpHeaders();
-        scimRequestHeaders.set("Authorization", "Bearer " + tokenResponse.get("access_token"));
-
-        String scimResponse = null;
-
+        Client client = clientBuilder.build();
+        WebTarget target = client.target(identityServerUrl).path("/oauth/token").queryParam("grant_type", "client_credentials");
```

Comment:
I think this one is already implemented in IdentityClient.java (or at least very similar), I would prefer  to extend that one